vaccine-standby/functions/login.js

```diff
@@ -0,0 +1,34 @@
+const crypto = require('crypto');
+
+// eslint-disable-next-line func-names
+exports.handler = function (context, event, callback) {
+  // Create a token from the password, and use it to check by setting it
+  function createToken(password) {
+    const tokenString = `${context.ACCOUNT_SID}:${password}:${context.SALT}`;
+
+    return crypto
+      .createHmac('sha1', context.AUTH_TOKEN)
+      .update(Buffer.from(tokenString, 'utf-8'))
+      .digest('base64');
+  }
+
+  function isAllowed(token) {
+    // Create the token with the environment password
+    const masterToken = createToken(context.ADMIN_PASSWORD);
+    return masterToken === token;
+  }
```

Comment:
The `createToken` and `isAllowed` functions could be extracted to a private function and then used in both this function and `get-studio-executions.js`.

Reply:
To add to this, you can enable a function to be set as private by including it in its name for example `login.private.js`

Default without it is to be set to public, but you could also set it to be `protected`

Here's the docs on it https://www.twilio.com/docs/runtime/functions-assets-api/api/understanding-visibility-public-private-and-protected-functions-and-assets

The convention of having the visibility in the filename itself prior to the extension is unique to the Quick Deploy service

Reply:
This has been addressed with the creation of the `auth` private function